### PR TITLE
Added dev guidelines for azure actions

### PR DESCRIPTION
**Pull request by BALAGA-GAYATRI**
This is based on azure login action's dev cycle. The steps are written in a generic way so that it is applicable for the other azure actions as well. In those actions this pointer for guidance will be added next.

### CODE REVIEW

File: docs/developer-guildelines.md
Comment (kaverma) [inline docs/developer-guildelines.md:1]: We should add a link to this in the TOC in the readMe for this repo as well.

File: README.md
Comment (kaverma) [inline README.md:11]: typo - "existing  existing"

File: docs/developer-guildelines.md
Comment (kaverma) [inline docs/developer-guildelines.md:13]: Please clarify this step more. Where to create a .yml workflow for testing - In a test repo?

File: docs/developer-guildelines.md
Comment (kaverma) [inline docs/developer-guildelines.md:17]: We should add a separate action on Releasing the changes also. After the PR is merged, taking the changes to release branch, testing on that branch and then updating the version tag.

File: docs/developer-guildelines.md
Comment (kaverma) [inline docs/developer-guildelines.md:9]: nit: we can remove 'significant'. 'If there are any changes in input and .."

File: README.md
Comment (BALAGA-GAYATRI) [reply README.md]: fixed it. Thanks for pointing out!

**Review by kaverma — CHANGES_REQUESTED** [review]
@BALAGA-GAYATRI - Added few comments, can you please have a look.

File: docs/developer-guildelines.md
Comment (ds-ms) [inline docs/developer-guildelines.md:3]: to the local `main` branch.

File: docs/developer-guildelines.md
Comment (ds-ms) [inline docs/developer-guildelines.md:5]: The numbering seems to be wrong through out.

Comment (ds-ms) [inline docs/developer-guildelines.md:5]: Before running `npm install` mention the minimum required versions of node and npm to be present in the machine

File: docs/developer-guildelines.md
Comment (ds-ms) [inline docs/developer-guildelines.md:11]: You're supposed to check in `node_modules` only in `releases/v*` branches

File: docs/developer-guildelines.md
Comment (ds-ms) [reply docs/developer-guildelines.md:13]: Remove the section about publish profile, it's only specific to `web-apps`

File: docs/developer-guildelines.md
Comment (ds-ms) [inline docs/developer-guildelines.md:15]: `main` branch

File: docs/developer-guildelines.md
Comment (kaverma) [inline docs/developer-guildelines.md:26]: Once your main branch PR is merged, cherry pick your changes to __releases/v(latest)__ branch and create a PR against the releases branch.

Comment (kaverma) [inline docs/developer-guildelines.md:26]: Please move this part to next step -
10. After that release PR is merged the repo maintainer will create a new release with a new tag. This release will be pre-release mode until certain bug bashes/regressive testing is done. Here for __releases/v(latest)__ branch, checkin your __node_modules__ if there is a new dependency in package.json.

**Review by kaverma — APPROVED** [review]

**Pull request merged**
